sdk/eventhub/event-hubs/src/eventHubClient.ts

```diff
@@ -171,58 +178,85 @@ export class EventHubClient {
    */
   constructor(
     host: string,
-    entityPath: string,
+    eventHubPath: string,
     credentials: ApplicationTokenCredentials | UserTokenCredentials | DeviceTokenCredentials | MSITokenCredentials,
     options?: ClientOptions
   );
   constructor(
-    host: string,
-    entityPath: string,
-    tokenProviderOrCredentials:
+    hostOrConnectionString: string,
+    eventHubPathOrOptions?: string | ClientOptions,
+    tokenProviderOrCredentials?:
       | TokenProvider
       | ApplicationTokenCredentials
       | UserTokenCredentials
       | DeviceTokenCredentials
       | MSITokenCredentials,
     options?: ClientOptions
   ) {
-    if (!host) {
-      throw new Error(
-        "Please provide a fully qualified domain name of your Event Hub instance for the 'host' parameter."
+    let connectionString;
+    let sharedAccessKeyName = "defaultKeyName";
+    let sharedAccessKey = "defaultKeyValue";
+    let eventHubPath: string | undefined;
+    let tokenProvider = {} as TokenProvider;
+
+    hostOrConnectionString = String(hostOrConnectionString);
+
+    if (hostOrConnectionString.startsWith("Endpoint")) {
+      connectionString = hostOrConnectionString;
+      const parsedCS = parseConnectionString<ServiceBusConnectionStringModel>(
+        connectionString
       );
+      if (!parsedCS.EntityPath){
+        throw new TypeError(
+          `"connectionString": "${connectionString}", ` +
+            `must contain EntityPath="<path-to-the-entity>".`
+        );
+      }
+    } else {
+      let host = hostOrConnectionString;
+      if (!host) {
+        throw new Error(
+          "Please provide a fully qualified domain name of your Event Hub instance for the 'host' parameter."
+        );
+      }
+      if (!host.endsWith("/")) host += "/";
+      connectionString = `Endpoint=sb://${host};SharedAccessKeyName=${sharedAccessKeyName};SharedAccessKey=${sharedAccessKey};`;
     }
 
-    if (!entityPath) {
-      throw new Error("Please provide a value for the 'entityPath' parameter.");
-    }
+    if (typeof eventHubPathOrOptions === "string") {
```

Comment:
If user provides host and then options instead of event hub path, we should throw error, but we arent doing so here